code/datums/components/mob_harvest.dm

```diff
@@ -0,0 +1,157 @@
+ */
+/obj/item/mob_harvest
+	name = "magical mob-powered item generator"
+	desc = "put inside mob to magically make items. you probably shouldn't be able to see this"
+	///typepath of the item you want to harvest
+	var/produced_item_typepath = /obj/item/food/bait
+	///stand-in name for the item
+	var/produced_item_desc = "funny worm"
+	///how much is ready to harvest
+	var/amount_ready = 1
+	///max amount that can be stored between harvests
+	var/max_ready = 1
+	///mob that has the component
+	var/mob/living/harvest_mob
+	///time between item creation
+	var/item_generation_wait = 10 SECONDS
+	///tracked time
+	var/item_generation_time = 0
+	///time to reduce when fed
+	var/item_reduction_time = 2 SECONDS
+
+/obj/item/mob_harvest/Initialize(mapload, harvest_mob)
+	src.harvest_mob = harvest_mob
+	item_generation_time = item_generation_wait
+	START_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/item/mob_harvest/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
+	harvest_mob = null
+
+/obj/item/mob_harvest/process(delta_time)
+	///only track time if we aren't dead and have room for more items
+	if(harvest_mob.stat != DEAD && amount_ready < max_ready)
+		item_generation_time -= delta_time
+		if(item_generation_time <= 0)
+			item_generation_time = item_generation_wait
+			amount_ready++
+
+/**
+ * Proc called from attacking the component parent with the correct item, reduces wait time between items
+ *
+ * Arguments:
+ * * mob/user - who is trying to do this
+ */
+/obj/item/mob_harvest/proc/remove_wait_time(mob/user)
+	if(amount_ready >= max_ready)
+		to_chat(user, span_warning("[harvest_mob] looks too full to keep feeding!"))
+		return		
+	item_generation_time -= item_reduction_time
+	to_chat(user, span_notice("You feed [harvest_mob]."))
+	return
+
+/**
+ * Proc called from attacking the component parent with the correct item, handles creating the item
+ *
+ * Arguments:
+ * * mob/user - who is trying to do this
+ */
+/obj/item/mob_harvest/proc/harvest_item(mob/user)
+	if(amount_ready >= 1)
+		to_chat(user, span_notice("You harvest some [produced_item_typepath] from [harvest_mob]."))
+		amount_ready--
+		new produced_item_typepath(get_turf(harvest_mob))
+		return
+	else
+		to_chat(user, span_warning("[harvest_mob] doesn't seem ready yet to harvest from."))
+
+/**
+ * # Sheep's mob_harvest item
+ *
+ * Used by sheep to generate wool. Has an extra check for amount ready to determine if sheep looks sheared or not.
+ */
+/obj/item/mob_harvest/sheep
+	name = "fluffy wool generator"
+	desc = "creates wool through sheer will and determination"
+	produced_item_typepath = /obj/item/stack/sheet/cotton/wool
+	produced_item_desc = "soft wool"
+	item_generation_wait = 5 MINUTES
+	item_reduction_time = 30 SECONDS
+	var/woolless = FALSE
+
+/obj/item/mob_harvest/sheep/process(delta_time)
+	..()
+	if(amount_ready > 0 && woolless)
+		woolless = FALSE
+		harvest_mob.icon_state = "sheep"
```

Comment:
This would probably work a bit nicer if you send a signal whenever the harvest status of the animal changes. The mob would register to this signal so that the mob itself can decide what to do when it becomes harvest-able, instead of changing the mobs icon_state in this object.